CPDShell/Core/algorithms/ClassificationBasedCPD/test_statistics/threshold_overcome.py

```diff
@@ -0,0 +1,31 @@
+"""
+Module for implementation of test statistic based on threshold overcome.
+"""
+
+__author__ = "Artemii Patov"
+__copyright__ = "Copyright (c) 2024 Artemii Patov"
+__license__ = "SPDX-License-Identifier: MIT"
+
+from CPDShell.Core.algorithms.ClassificationBasedCPD.abstracts.istatistic_test import TestStatistic
+
+
+class ThresholdOvercome(TestStatistic):
+    """
+    The class implementing test statistic based on threshold overcome.
+    """
+
+    def __init__(self, threshold: float) -> None:
+        """
+        Initializes a new instance of threshold overcome criterion.
```

Comment:
Maybe you'd be better to consult with our supervisor here. As far as I remember, he said that the word “test” is more often used in this context than “criterion”.